optuna/pruners/hyperband.py

```diff
@@ -172,6 +202,7 @@ def _calculate_trial_allocation_budget(self, pruner_index: int) -> int:
         number of trials here instead.
         """
 
+        assert isinstance(self._n_brackets, int)
```

Comment:
I don't think this necessary.

ref: https://docs.python.org/3/library/math.html#math.floor

Reply:
This assertion is needed for the type checking. If this statement is removed, the following error occurs with `mypy .`.
```
optuna/pruners/hyperband.py:204: error: Unsupported operand types for - ("None" and "int")
optuna/pruners/hyperband.py:204: note: Left operand is of type "Optional[int]"
```